src/coreclr/vm/threadsuspend.cpp

```diff
@@ -2912,9 +2901,23 @@ BOOL Thread::RedirectThreadAtHandledJITCase(PFN_REDIRECTTARGET pTgt)
 
     //////////////////////////////////////
     // Get and save the thread's context
+    BOOL bRes = true;
 
     // Always get complete context, pCtx->ContextFlags are set during Initialization
-    BOOL bRes = EEGetThreadContext(this, pCtx);
+
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
+    // Scenarios like GC stress may indirectly disable XState features in the pCtx
+    // depending on the state at the time of GC stress interrupt.
+    // 
+    // Make sure that AVX feature mask is set, if supported.
+    // 
+    // This should not normally fail.
+    // The system silently ignores any feature specified in the FeatureMask
+    // which is not enabled on the processor.
+    bRes &= SetXStateFeaturesMask(pCtx, XSTATE_MASK_AVX);
```

Comment:
I only have one nit - I would prefer failures / asserts on each of the failure instead of combining the results together.

Reply:
`GetXStateFeaturesMask` and `SetXStateFeaturesMask` basically cannot fail. They are very tolerant to the input. 
To get failures you'd need something like pass flags for a wrong architecture (like arm64). As-is that would not compile.

Realistically only `CopyContext` is something that could fail here due to some mistake.
I think in this particular case individual asserts/returns would not add a lot of value.

Reply:
I've just found those would fail also when passed context without any XSTATE present. It seems like something that could possibly happen for the usage in `Thread::RedirectCurrentThreadAtHandledJITCase` where we pass it a general context that we get from the vectored exception handler. So it seems that it would be nice to actually make that work in that case.

Reply:
Hmm, my read on that was that Set(Get) would be a noop, but I think you are right, if no XState is set it may fail. Arguably a bug in the validation of Set not accepting 0.

Yes, we should make this case work.